### ISSUE TITLE
Python 3.9.0a6: "PendingDeprecationWarning: lib2to3 package is deprecated" breaks the tests

### ISSUE THREAD

**Issue by hroncok** [pypa/setuptools#2081]
The setuptools test suite errors with new PendingDeprecationWarnings about lib2to3.

```
+ PYTHONPATH=/builddir/build/BUILD/setuptools-46.1.3
+ pytest-3.9 --ignore=pavement.py
============================= test session starts ==============================
platform linux -- Python 3.9.0a6, pytest-4.6.9, py-1.8.0, pluggy-0.13.0
rootdir: /builddir/build/BUILD/setuptools-46.1.3, inifile: pytest.ini
plugins: virtualenv-1.7.0, shutil-1.7.0
collected 675 items / 9 errors / 2 skipped / 664 selected

==================================== ERRORS ====================================
__________________ ERROR collecting setuptools/lib2to3_ex.py ___________________
setuptools/lib2to3_ex.py:12: in <module>
    from lib2to3.refactor import RefactoringTool, get_fixers_from_package
/usr/lib64/python3.9/lib2to3/__init__.py:4: in <module>
    warnings.warn(
E   PendingDeprecationWarning: lib2to3 package is deprecated and may not be able to parse Python 3.10+
_______________ ERROR collecting setuptools/command/build_py.py ________________
setuptools/command/build_py.py:16: in <module>
    from setuptools.lib2to3_ex import Mixin2to3
setuptools/lib2to3_ex.py:12: in <module>
    from lib2to3.refactor import RefactoringTool, get_fixers_from_package
/usr/lib64/python3.9/lib2to3/__init__.py:4: in <module>
    warnings.warn(
E   PendingDeprecationWarning: lib2to3 package is deprecated and may not be able to parse Python 3.10+
_________________ ERROR collecting setuptools/command/test.py __________________
setuptools/command/test.py:18: in <module>
    from .build_py import _unique_everseen
setuptools/command/build_py.py:16: in <module>
    from setuptools.lib2to3_ex import Mixin2to3
setuptools/lib2to3_ex.py:12: in <module>
    from lib2to3.refactor import RefactoringTool, get_fixers_from_package
/usr/lib64/python3.9/lib2to3/__init__.py:4: in <module>
    warnings.warn(
E   PendingDeprecationWarning: lib2to3 package is deprecated and may not be able to parse Python 3.10+
______________ ERROR collecting setuptools/tests/test_develop.py _______________
setuptools/tests/test_develop.py:14: in <module>
    from setuptools.command import test
setuptools/command/test.py:18: in <module>
    from .build_py import _unique_everseen
setuptools/command/build_py.py:16: in <module>
    from setuptools.lib2to3_ex import Mixin2to3
setuptools/lib2to3_ex.py:12: in <module>
    from lib2to3.refactor import RefactoringTool, get_fixers_from_package
/usr/lib64/python3.9/lib2to3/__init__.py:4: in <module>
    warnings.warn(
E   PendingDeprecationWarning: lib2to3 package is deprecated and may not be able to parse Python 3.10+
______________ ERROR collecting setuptools/tests/test_develop.py _______________
setuptools/tests/test_develop.py:14: in <module>
    from setuptools.command import test
setuptools/command/test.py:18: in <module>
    from .build_py import _unique_everseen
setuptools/command/build_py.py:16: in <module>
    from setuptools.lib2to3_ex import Mixin2to3
setuptools/lib2to3_ex.py:12: in <module>
    from lib2to3.refactor import RefactoringTool, get_fixers_from_package
/usr/lib64/python3.9/lib2to3/__init__.py:4: in <module>
    warnings.warn(
E   PendingDeprecationWarning: lib2to3 package is deprecated and may not be able to parse Python 3.10+
_____________ ERROR collecting setuptools/tests/test_namespaces.py _____________
setuptools/tests/test_namespaces.py:9: in <module>
    from setuptools.command import test
setuptools/command/test.py:18: in <module>
    from .build_py import _unique_everseen
setuptools/command/build_py.py:16: in <module>
    from setuptools.lib2to3_ex import Mixin2to3
setuptools/lib2to3_ex.py:12: in <module>
    from lib2to3.refactor import RefactoringTool, get_fixers_from_package
/usr/lib64/python3.9/lib2to3/__init__.py:4: in <module>
    warnings.warn(
E   PendingDeprecationWarning: lib2to3 package is deprecated and may not be able to parse Python 3.10+
_____________ ERROR collecting setuptools/tests/test_namespaces.py _____________
setuptools/tests/test_namespaces.py:9: in <module>
    from setuptools.command import test
setuptools/command/test.py:18: in <module>
    from .build_py import _unique_everseen
setuptools/command/build_py.py:16: in <module>
    from setuptools.lib2to3_ex import Mixin2to3
setuptools/lib2to3_ex.py:12: in <module>
    from lib2to3.refactor import RefactoringTool, get_fixers_from_package
/usr/lib64/python3.9/lib2to3/__init__.py:4: in <module>
    warnings.warn(
E   PendingDeprecationWarning: lib2to3 package is deprecated and may not be able to parse Python 3.10+
________________ ERROR collecting setuptools/tests/test_test.py ________________
setuptools/tests/test_test.py:11: in <module>
    from setuptools.command.test import test
setuptools/command/test.py:18: in <module>
    from .build_py import _unique_everseen
setuptools/command/build_py.py:16: in <module>
    from setuptools.lib2to3_ex import Mixin2to3
setuptools/lib2to3_ex.py:12: in <module>
    from lib2to3.refactor import RefactoringTool, get_fixers_from_package
/usr/lib64/python3.9/lib2to3/__init__.py:4: in <module>
    warnings.warn(
E   PendingDeprecationWarning: lib2to3 package is deprecated and may not be able to parse Python 3.10+
________________ ERROR collecting setuptools/tests/test_test.py ________________
setuptools/tests/test_test.py:11: in <module>
    from setuptools.command.test import test
setuptools/command/test.py:18: in <module>
    from .build_py import _unique_everseen
setuptools/command/build_py.py:16: in <module>
    from setuptools.lib2to3_ex import Mixin2to3
setuptools/lib2to3_ex.py:12: in <module>
    from lib2to3.refactor import RefactoringTool, get_fixers_from_package
/usr/lib64/python3.9/lib2to3/__init__.py:4: in <module>
    warnings.warn(
E   PendingDeprecationWarning: lib2to3 package is deprecated and may not be able to parse Python 3.10+
=========================== short test summary info ============================
SKIPPED [2] /builddir/build/BUILD/setuptools-46.1.3/setuptools/tests/test_msvc.py:18: could not import 'distutils.msvc9compiler': No module named 'winreg'
!!!!!!!!!!!!!!!!!!! Interrupted: 9 errors during collection !!!!!!!!!!!!!!!!!!!!
====================== 2 skipped, 9 error in 1.75 seconds ======================
```

**Comment by hroncok** [comment]
Working on a PR that filter those.

**Comment by hroncok** [comment]
https://github.com/pypa/setuptools/pull/2082

**Comment by jaraco** [comment]
As written, this bug report is only about the tests failing. What we really need is a ticket tracking the _effect_ of the warning. That is, this package is being warned that something is going to break. The project probably needs to take action and not just suppress the alert.

**Comment by hroncok** [comment]
I agree. Shall I open the long term ticket?

**Comment by jaraco** [comment]
Yes please.

**Comment by hroncok** [comment]
https://github.com/pypa/setuptools/issues/2086

**Comment by jaraco** [comment]
Fixed in #2082. Thanks for reporting and for the follow-up action!
